### PR TITLE
Default to THROUGHPUT_BASED autoscaling for classic templates

### PR DESCRIPTION
**Pull request by damccorm**
Before migrating the release process to GitHub, most streaming classic templates had autoscaling enabled by default. This functionality was lost as part of the move over, and this seeks to restore that default.

This has caused user pain because there's not an easy way to enable autoscaling for classic templates if its not on by default.

Note that I chose 10 workers as the default to avoid overwhelming users projects with high scaling (this happened to our templates project with integration tests when I tried 100) and because it seems like a reasonable default. It will cause streaming engine jobs without a default set to change their behavior (currently they default to 100 max workers)

### CODE REVIEW

File: plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java [hunk 1]
```diff
@@ -258,6 +258,11 @@ protected String stageClassicTemplate(
     arguments.add(element("argument", "--templateLocation=" + templatePath));
     arguments.add(element("argument", "--project=" + projectId));
     arguments.add(element("argument", "--region=" + region));
+    if (imageSpec.getMetadata().isStreaming()) {
+      // Default to THROUGHPUT_BASED autoscaling for streaming classic templates
```
Comment (liferoad) [inline plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java:262]: so the user cannot change these settings?

Comment (damccorm) [reply plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java:262]: They can adjust maxNumWorkers, but not the autoscaling algorithm

Comment (liferoad) [reply plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java:262]: will `arguments.add(element("argument", "--maxNumWorkers=10"));` override the maxNumWorkers option specified by the user?

Comment (damccorm) [reply plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java:262]: Overriding maxNumWorkers still works. I updated to demonstrate this with a test below which produced this job - https://pantheon.corp.google.com/dataflow/jobs/us-central1/2024-04-04_09_33_22-17280287866962791451;graphView=0?project=cloud-teleport-testing&e=13802955&mods=logs_tg_staging&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22))

**Review by an2x — DISMISSED** [review]

**Review by liferoad — APPROVED** [review]

**Pull request merged**